api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java

```diff
@@ -51,16 +73,15 @@ public boolean isInitialized() {
 	 * @should throw an error if generated identifier is longer than maxLength
      */
     public String getIdentifierForSeed(long seed) {
-
     	// Convert the next sequence integer into a String with the appropriate Base characters
 		int seqLength = firstIdentifierBase == null ? 1 : firstIdentifierBase.length();
 
 		String identifier = IdgenUtil.convertToBase(seed, baseCharacterSet.toCharArray(), seqLength);
-
-    	// Add optional prefix and suffix
-    	identifier = (prefix == null ? identifier : prefix + identifier);
+		
+		PrefixProvider prefixProvider = getPrefixProvider(prefix);
+		identifier = prefixProvider.getValue() + identifier;
```

Comment:
@mseaton the prefix is always being dealt as a configuration, defaulting to `ConstantPrefixProvider` in most cases.